includes/Story_Post_Type.php

```diff
@@ -152,6 +155,54 @@ public function init() {
 		}
 	}
 
+	/**
+	 * Adds story capabilities to default user roles.
+	 *
+	 * This gives WordPress site owners more granular control over story management,
+	 * as they can customize this to their liking.
+	 *
+	 * @return void
+	 */
+	public function add_caps_to_roles() {
+		$post_type_object = get_post_type_object( self::POST_TYPE_SLUG );
+
+		if ( ! $post_type_object ) {
+			return;
+		}
+
+		$all_capabilities = array_values( (array) $post_type_object->cap );
+
+		$administrator = get_role( 'administrator' );
+		$editor        = get_role( 'editor' );
+		$author        = get_role( 'author' );
+		$contributor   = get_role( 'contributor' );
+
+		if ( $administrator instanceof WP_Role ) {
+			foreach ( $all_capabilities as $cap ) {
+				$administrator->add_cap( $cap );
+			}
+		}
+
+		if ( $editor instanceof WP_Role ) {
+			foreach ( $all_capabilities as $cap ) {
+				$editor->add_cap( $cap );
+			}
+		}
+
+		if ( $author instanceof WP_Role ) {
+			$author->add_cap( 'edit_web-stories' );
+			$author->add_cap( 'edit_published_web-stories' );
+			$author->add_cap( 'delete_web-stories' );
+			$author->add_cap( 'delete_published_web-stories' );
+			$author->add_cap( 'publish_web-stories' );
+		}
+
+		if ( $contributor instanceof WP_Role ) {
+			$contributor->add_cap( 'edit_web-stories' );
+			$contributor->add_cap( 'delete_web-stories' );
+		}
```

Comment:
```suggestion
         $roles = [  'administrator', 'editor', 'author', 'contributor' ];
         $roles = apply_filters( 'web_stories_roles', $roles );
         foreach( $roles as $_role ) {
                   $role = get_role( $_role );
                   if ( !( $role instanceof WP_Role ) ) {
                      continue;
                   }
                   
            if ( 'author' === $_role ) {
			   $role->add_cap( 'edit_web-stories' );
			   $role->add_cap( 'edit_published_web-stories' );
			   $role->add_cap( 'delete_web-stories' );
			   $role->add_cap( 'delete_published_web-stories' );
			   $role->add_cap( 'publish_web-stories' );
		    } else if ( 'contributor' === $_role ) {
			   $role->add_cap( 'edit_web-stories' );
			   $role->add_cap( 'delete_web-stories' );
		     } else  {
			   foreach ( $all_capabilities as $cap ) {
			       $role->add_cap( $cap );
			   }
		     }
         }
```

Reply:
Refactor to make a little cleaner and add a filter.

Reply:
Hmm I feel like this makes it actually less readable.

Also, the filter is dangerous. If I add `subscriber` to the list, that role will get all capabilities. An action would be more appropriate if people want to modify the roles. But then again, they could already do that on `init` or similar.

Reply:
I think maybe an action is better than the filter then. I personally found this more readable, but I don't feel so strongly about it.